histogram.go

```diff
@@ -0,0 +1,42 @@
+package imaging
+
+import (
+	"image"
+)
+
+// Histogram returns a normalized histogram of an image.
+//
+// Resulting histogram is represented as an array of 256 floats, where
+// histogram[i] is a probability of a pixel being of a particular luminance i.
+func Histogram(img image.Image) [256]float32 {
```

Comment:
> Maybe it's better to use [256]float64 as a return value? Go's math package operates on float64, so it will be easier to further use the histogram (less explicit type conversions). Not sure about it.

Fair point!